detection_rules/rule.py

```diff
@@ -442,6 +443,23 @@ class TOMLRuleContents(BaseRuleContents, MarshmallowDataclassMixin):
     """Rule object which maps directly to the TOML layout."""
     metadata: RuleMeta
     data: AnyRuleData = field(metadata=dict(data_key="rule"))
+    # _version_lock: Optional[Any] = None
+
+    @cached_property
+    def version_lock(self):
+        # VersionLock
+        from .version_lock import default_version_lock
+
+        return getattr(self, '_version_lock', None) or default_version_lock
+
+    def set_version_lock(self, value):
+        from .version_lock import VersionLock
+
+        if value and not isinstance(value, VersionLock):
+            raise TypeError(f'version lock property must be set with VersionLock objects only. Got {type(value)}')
+
+        # circumvent frozen class
+        self.__dict__['_version_lock'] = value
```

Comment:
using `cached_property` with ☝️ this allows us to set a property in a frozen dataclass